internal/cortex/frontend/transport/handler.go

```diff
@@ -48,6 +48,7 @@ type HandlerConfig struct {
 	QueryStatsEnabled        bool          `yaml:"query_stats_enabled"`
 	LogFailedQueries         bool          `yaml:"log_failed_queries"`
 	FailedQueryCacheCapacity int           `yaml:"failed_query_cache_capacity"`
+	FailedQueryTTL           time.Duration `yaml:"failed_query_ttl"`
```

Comment:
You need to add a cmd line flag for this config.
<img width="1095" alt="image" src="https://github.com/user-attachments/assets/959ee0b7-7ff9-4e98-9dea-05a4d48fa2f8">